Automattic/wp-calypso
# Use API endpoints

#### Changes proposed in this Pull Request

- **The counts represent unseen posts since the user subscribed to the feed or this feature's "release date" (07/13/2020) whichever is newer.**
- **For now a post is marked as seen when visited in the Reader or if a "Mark as seen" button is used**
- **Seen status should be persisted**


If there is unseen content in your reader the dot should be visible, currently it will update with new status only on page reload.

![Markup on 2020-06-12 at 10:01:36 PM](https://user-images.githubusercontent.com/2129455/84537593-ba8ea980-acf8-11ea-8765-6d66fbd978d4.png)

The  Reader sections should present **unseen counts** for sections and particular blogs. 

![Markup on 2020-06-12 at 10:01:52 PM](https://user-images.githubusercontent.com/2129455/84537591-ba8ea980-acf8-11ea-9863-6e07f7a9b806.png)

Sections will have "Mark all as seen" buttons, active only if there is unseen content in the feed.

![Markup on 2020-06-12 at 10:02:31 PM](https://user-images.githubusercontent.com/2129455/84537590-b9f61300-acf8-11ea-94ac-24f143c0f468.png)

Feeds will have "Mark all as seen" buttons, active only if there is unseen content in the feed.

![Markup on 2020-06-12 at 10:03:24 PM](https://user-images.githubusercontent.com/2129455/84537579-b82c4f80-acf8-11ea-9ae2-07cbe949d609.png)

Posts and combined posts will have "Mark as seen/unseen" buttons

![Markup on 2020-06-12 at 10:03:08 PM](https://user-images.githubusercontent.com/2129455/84537583-b8c4e600-acf8-11ea-8a76-4e7a40b5888a.png)
![Markup on 2020-06-12 at 10:02:46 PM](https://user-images.githubusercontent.com/2129455/84537588-b9f61300-acf8-11ea-9a00-ba9fb30a8484.png)

When visiting a post in the Reader there will be a seen icon

![Markup on 2020-06-12 at 10:04:15 PM](https://user-images.githubusercontent.com/2129455/84537571-b5315f00-acf8-11ea-8de3-2e74b8bf214f.png)

#### Testing instructions
- Apply D46339-code
- Unseen dot should be visible in main bar
- Unseen counts should be visible in Reader
- Mark as seen/unseen single/combined feed items (counts should be updated)
- Mark as seen entire feed (counts should be updated)
- Mark as seen entire sections like Following/All (counts should be updated)
- Visit a full post in Reader (should mark it as seen)
